src/restricted-endpoints/document/document.controller.spec.ts

```diff
@@ -12,11 +12,18 @@ import { CouchdbService } from '../../couchdb/couchdb.service';
 import { of, throwError } from 'rxjs';
 import { DocumentRule } from '../../permissions/rules/rules.service';
 import { UnauthorizedException } from '@nestjs/common';
+import { Request as Req } from 'express';
 
 describe('DocumentController', () => {
   let controller: DocumentController;
   let mockPermissionService: PermissionService;
   let mockCouchDBService: CouchdbService;
+
+  const headers = {};
+  const mockReq = {
```

Comment:
can we add a test case checking for the new functionality also, please? replication-backend is a production system and key service for us :blush: